app/models/host.rb

```diff
@@ -198,35 +198,33 @@ class Host < ApplicationRecord
   supports_not :vmotion_enabled
   supports     :ipmi do
     if ipmi_address.blank?
-      unsupported_reason_add(:ipmi, _("The Host is not configured for IPMI"))
+      _("The Host is not configured for IPMI")
     elsif authentication_type(:ipmi).nil?
-      unsupported_reason_add(:ipmi, _("The Host has no IPMI credentials"))
+      _("The Host has no IPMI credentials")
     elsif authentication_userid(:ipmi).blank? || authentication_password(:ipmi).blank?
-      unsupported_reason_add(:ipmi, _("The Host has invalid IPMI credentials"))
+      _("The Host has invalid IPMI credentials")
     end
   end
 
   supports :start do
-    valid_states = respond_to?(:vim_power_up_from_standby) ? %w[off standby] : "off"
+    valid_states = respond_to?(:vim_power_up_from_standby) ? %w[off standby] : %w[off]
 
     if !supports?(:ipmi)
-      unsupported_reason_add(:start, unsupported_reason(:ipmi))
-    elsif Array(valid_states).exclude?(power_state)
-      unsupported_reason_add(:start, _("The Host is not in power state #{valid_states.inspect}"))
+      unsupported_reason(:ipmi)
+    elsif valid_states.exclude?(power_state)
+      _("The Host is not in power state #{valid_states.inspect}")
     end
   end
 
   supports :stop do
     if !supports?(:ipmi)
-      unsupported_reason_add(:stop, unsupported_reason(:ipmi))
+      unsupported_reason(:ipmi)
     elsif power_state != "on"
-      unsupported_reason_add(:stop, _("The Host is not in powered on"))
+      _("The Host is not in powered on")
     end
   end
 
-  supports :reset do
-    unsupported_reason_add(:reset, unsupported_reason(:ipmi)) unless supports?(:ipmi)
-  end
+  supports(:reset) { unsupported_reason(:ipmi) }
```

Comment:
was the `unless supports?(:ipmi)` dropped intentionally?

Reply:
Yea. Once looking into this feature it started morphing on its own.
Love it when this happens.

```ruby
unsupported_reason_add(:reset, unsupported_reason(:ipmi)) unless supports?(:ipmi)
## =>
unsupported_reason(:ipmi) unless supports?(:ipmi)
## =>
if supports?(:ipmi)
  nil
else
  unsupported_reason(:ipmi)
end
## =>
if supports?(:ipmi)
  unsupported_reason(:ipmi) # nil
else
  unsupported_reason(:ipmi)
end
## =>
 unsupported_reason(:ipmi)
```